src/app/clusters/webrtc-transport-provider-server/webrtc-transport-provider-server.cpp

```diff
@@ -226,23 +226,31 @@ WebRTCSessionStruct * WebRTCTransportProviderServer::CheckForMatchingSession(Han
 
 uint16_t WebRTCTransportProviderServer::GenerateSessionId()
```

Comment:
How about changing the signature to `CHIP_ERROR WebRTCTransportProviderServer::GenerateSessionId(uint16_t & id)` ? Then instead of aborting you could return appropriate error code so the caller can wait some time and try to get session ID again.

Reply:
> there is no any recovery mechanism defined in the spec after running out of sessionID

So maybe the spec should be somehow updated. Crashing entire application/device because one session could not be established seems strange.